src/main/scala/zio/s3/S3.scala

```diff
@@ -0,0 +1,395 @@
+      Files.exists(path / bucketName)
+
+    override def listBuckets(): ZIO[Blocking, S3Exception, S3BucketListing] =
+      Files
+        .list(path)
+        .filterM(p => Files.readAttributes[PosixFileAttributes](p).map(_.isDirectory))
+        .mapM { p =>
+          Files
+            .readAttributes[PosixFileAttributes](p)
+            .map(attr => S3Bucket(p.filename.toString, attr.creationTime().toInstant))
+        }
+        .runCollect
+        .mapError(S3ExceptionLike)
+        .map(S3BucketListing(_))
+
+    override def deleteObject(bucketName: String, key: String): ZIO[Blocking, S3Exception, Unit] =
+      Files.delete(path / bucketName / key).mapError(S3ExceptionLike(_))
+
+    override def getObject(bucketName: String, key: String): ZStreamChunk[Blocking, S3Exception, Byte] = ZStreamChunk {
+      ZStream
+        .managed(ZManaged.fromAutoCloseable(Task(new FileInputStream((path / bucketName / key).toFile))))
+        .flatMap(ZStream.fromInputStream(_, 2048).chunks)
+        .mapError(S3ExceptionLike)
+    }
+
+    override def listObjects(
+      bucketName: String,
+      prefix: String,
+      maxKeys: Int
+    ): ZIO[Blocking, S3Exception, S3ObjectListing] =
+      Files
+        .find(path / bucketName) { (p, _) =>
+          p.filename.toString().startsWith(prefix)
+        }
+        .take(maxKeys)
+        .map(_.filename.toString())
+        .runCollect
+        .map(_.map(S3ObjectSummary(bucketName, _)))
+        .map(S3ObjectListing(bucketName, _, None))
+        .mapError(S3ExceptionLike)
+
+    //TODO can we do better ???
+    override def getNextObjects(listing: S3ObjectListing): ZIO[Blocking, S3Exception, S3ObjectListing] =
+      ZIO.succeed(listing)
+
+    override def putObject[R1 <: Blocking](
+      bucketName: String,
+      key: String,
+      contentLength: Long,
+      contentString: String,
+      content: ZStreamChunk[R1, Throwable, Byte]
+    ): ZIO[R1, S3Exception, Unit] =
+      ZManaged
+        .fromAutoCloseable(Task(new FileOutputStream((path / bucketName / key).toFile)))
+        .use(os => content.run(ZSink.fromOutputStream(os)).unit)
+        .mapError(S3ExceptionLike)
+
+    override def execute[T](f: S3AsyncClient => CompletableFuture[T]): ZIO[Blocking, S3Exception, T] = ???
+
+    override def multipartUpload[R1 <: Blocking](n: Int)(
+      bucketName: String,
+      key: String,
+      contentType: String,
+      content: ZStream[R1, Throwable, Byte]
+    ): ZIO[R1, S3Exception, Unit] = putObject(bucketName, key, 0, contentType, content.chunkN(10))
+  }
+
+  //TODO remove when TestEnvironment is fixed
+  private val _ = new Test(_)
+
+// TODO environment problem since we have Blocking Context and
+//  object Test {
+//
+//    def connect(path: Path): ZManaged[Blocking, Nothing, S3] =
+//      ZManaged.succeed(new S3 {
+//        protected override val s3: Service[Any] = new Test(ZPath.fromJava(path))
+//      })
+//  }
+}
+
+case class S3BucketListing(buckets: List[S3Bucket])
```

Comment:
Maybe pull them out in separate files or to the companion, and remove `S3` prefix.